sqlglot/helper.py

```diff
@@ -7,40 +7,57 @@
 from copy import copy
 from enum import Enum
 
+from sqlglot.expressions import Expression, Table
+
 CAMEL_CASE_PATTERN = re.compile("(?<!^)(?=[A-Z])")
 logger = logging.getLogger("sqlglot")
 
 
 class AutoName(Enum):
-    def _generate_next_value_(name, _start, _count, _last_values):
+    def _generate_next_value_(name: str, _start: int, _count: int, _last_values: int) -> str:
         return name
 
 
-def list_get(arr, index):
+def list_get(arr: t.List[t.Any], index: int) -> t.Optional[t.Any]:
+    """Returns the value in `arr` at position `index`, or `None` if `index` is out of bounds."""
     try:
         return arr[index]
     except IndexError:
         return None
 
 
-def ensure_list(value):
+def ensure_list(value: t.Any) -> t.Union[t.List[t.Any], t.Tuple[t.Any], t.Set[t.Any]]:
```

Comment:
The proper thing to use here is [overload](https://docs.python.org/3/library/typing.html#typing.overload).